test/built-ins/Promise/prototype/finally/this-value-then-not-callable.js

```diff
@@ -0,0 +1,49 @@
+// Copyright (C) 2017 Jordan Harband. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+author: Jordan Harband
+description: >
+  Promise.prototype.finally called with a `this` value that does not define a
+  callable `then` property
+esid: sec-promise.prototype.finally
+features: [Symbol, Promise.prototype.finally, Function.prototype.call]
+---*/
+
+var symbol = Symbol();
+
+var p = new Promise(function () {});
+
+p.then = undefined;
+assert.throws(TypeError, function() {
+  Promise.prototype.finally.call(p);
```

Comment:
ah, interesting. sure, i can do that (no need to poison the finally method, but i can certainly pass a poisoned onFinally)